strawberryfields/gbs/sample.py

```diff
@@ -294,28 +284,76 @@ def to_subgraphs(samples: list, graph: nx.Graph) -> list:
     return subgraph_samples
 
 
-def gaussian(
+def vibronic(
+    t: np.ndarray,
     U1: np.ndarray,
     r: np.ndarray,
     U2: np.ndarray,
     alpha: np.ndarray,
     n_samples: int,
     loss: float = 0.0,
 ) -> list:
-    r"""Generate simulated samples from pure Gaussian states.
+    r"""Generate samples for computing vibronic spectra.
 
-    The pure Gaussian states are prepared by applying the following gates to the vacuum:
+    This function applies two procedures for generating samples depending on the temperature at
+    which the vibronic spectra is computed. In the finite temperature procedure, 2N vacuum modes
+    are prepared with N being the number of normal modes in the molecule and the first and
+    second N modes correspond to the normal modes of the final and initial electronic states,
+    respectively. These 2N modes are treated with two-mode squeezing gates and then the following
+    gates are applied to the first N modes:
 
     #. Interferometer ``U1``
-    #. Squeezing on all modes with parameters ``r``
+    #. Squeezing on all N modes with parameters ``r``
     #. Interferometer ``U2``
-    #. Displacement on all modes with parameters ``alpha``
+    #. Displacement on all N modes with parameters ``alpha``
+
+    In the last step, the number of photons in each of the 2N modes is measured.
+
+    In the zero temperature case, only N modes are considered. The four gates mentioned above are
+    applied to N vacuum states and then the number of photons in each mode is measured. This
+    makes the zero temperature sampling significantly faster than the finite temperature one.
+    However, to keep the sizes of the samples consistent and temperature independent, we add N
+    zeros to the end of each zero temperature sample at the very end of this function.
+
```

Comment:
It would be good to add a sentence here briefly explaining where the parameters in the example usage come from

Reply:
I will add it.